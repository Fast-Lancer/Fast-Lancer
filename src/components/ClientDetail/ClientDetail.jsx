import styles from './ClientDetail.module.css'
import phoneIcon from '../../assets/icons/phoneSmall.png'
import emailIcon from '../../assets/icons/emailSmall.png'
import paletteIcon from '../../assets/icons/paletteSmall.png'
import { Link } from 'react-router-dom'

export default function ClientDetail2({ client }) {
  const {
    client_name: name,
    email,
    phone,
    business_name: businessName,
    notes,
    projects
  } = client

  return (
    <main>
      <div className={styles.clientDetail}>
        <div className={styles.clientName}>{name}</div>
        <div className={styles.businessName}>
          <h1 aria-label="business-name">
            {businessName ?? 'No business name entered'}
          </h1>
          <h3>Business Name</h3>
        </div>
        <div className={styles.infoWrapper}>
          <h3>
            <div className={styles.contactDetails} aria-label="phone">
              <section className={styles.contactItem}><img src={phoneIcon} alt="phone icon" />
                {phone}
              </section>
              <section className={styles.contactItem}>
                <img src={emailIcon} alt="email" />
                {email}
              </section>
            </div>
          </h3>         

          <section className={styles.projectSection}> 
            <h2>
            Open Projects</h2>
            <section className={styles.projectList}>
              <ul>
                {
                  projects.map((project) => (
                    <li key={project.id}>
                      <Link to={`/projects/${project.id}`}>
                        <img src={paletteIcon} alt="palette" /> {project.title}
                      </Link>
                      <section className={styles.projectDescription}>
                        {project.description}
                      </section>
                    </li>
                  ))
                }
              </ul>
            </section>
          </section>
        </div>
        <section className={styles.notesSection}>
          <h2>Notes</h2>
          <p className={styles.notes} aria-label="notes">
            {notes ? notes : 'No notes entered.'}
          </p>
        </section>
      </div>
    </main>
  )
}
