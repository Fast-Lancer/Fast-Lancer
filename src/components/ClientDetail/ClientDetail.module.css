/*  Parent Wrapper  */
.clientDetail {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  background-color: #d3d3d3;
  font-size: 0.85em;
  width: 90vw;
}

.clientDetail a {
  display: flex;
  color: #595959;
  font-size: 1.25em;
}

.clientDetail a:hover {
  color: #fff;
  text-transform: none;
}

.clientName {
  font-family: "Playfair Display", serif;
  font-size: 4em;
  padding-bottom: 0.25em;
}

.businessName {
  width: 100%;
  height: fit-content;
  display: flex;
  flex-direction: column;
  align-content: center;
  align-items: center;
  font-family: "IBM Plex Mono", monospace;
  font-style: italic;
  font-size: 1em;
  padding-bottom: 1em;
  border-bottom: 1px solid black;
}

.businessName h1 {
  font-family: "IBM Plex Mono", monospace;
  font-size: 2.5em;
  font-style: normal;
  text-underline-offset: 0.5em;
  text-decoration: underline;
  padding-bottom: 0.5em;
  color: #595959;
}

.infoWrapper h3 {
  margin-top: 1vw;
  font-family: "IBM Plex Mono", monospace;
  font-size: 1.25em;
}

.contactDetails {
  display: grid;
  grid-template-columns: 1fr 1fr;
  justify-content: center;
  align-items: center;
}

.contactDetails img {
  padding: 0 1em 0 0;
  height: 1.2em;
}

.contactItem {
  display: flex;
  justify-content: center;
  align-items: center;
}

.projectSection {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 1em 0 1.2em 0;
  font-family: "IBM Plex Mono", monospace;
}

.projectSection h2 {
  font-size: 1.75em;
  padding-top: 1vw;
}

.projectSection img {
  padding-right: 1vw;
}

.projectList {
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  width: 30vw;
}

.projectList li {
  list-style: circle;
  margin-left: 0;
  padding: 1vw 0 0 0;
}

.projectDescription {
  font-style: italic;
}

.alignLeft {
  align-self: center;
  width: 100%;
  padding: 10px;
}

.alignLeft h2 {
  font-family: "Playfair Display", serif;
  font-size: 3em;
  font-weight: 800;
}

.textBlock {
  display: flex;
  align-items: center;
  padding-bottom: 1.5vw;
  margin: 1vw 0 1vw 1vw;
  border-bottom: 1px black solid;
  font-family: "IBM Plex Mono", sans-serif;
  font-weight: 600;
  font-size: 1.3em;
}

@media (max-width: 768px) {
  .infoSection {
    flex-direction: column;
  }

  .projectDetail {
    align-items: center;
  }
}
