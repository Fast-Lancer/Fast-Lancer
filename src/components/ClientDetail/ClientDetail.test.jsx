import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'
import ClientDetail from './ClientDetail.jsx'

it('should match the snapshot', () => {
  const { container } = render(
    <MemoryRouter initialEntries={['/']}>
      <ClientDetail client={{
        client_name: 'bob',
        email: 'bob@bob.com',
        phone: '123',
        business_name: 'bobs business',
        notes: 'yo bobs pretty cool',
        projects: [
          { id: 1, title: 'project1' },
          { id: 2, title: 'project2' }
        ]
      }} />
    </MemoryRouter>
  )

  expect(container).toMatchSnapshot()
})
