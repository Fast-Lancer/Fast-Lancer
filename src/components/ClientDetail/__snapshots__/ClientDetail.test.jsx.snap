// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<div>
  <main>
    <div
      class="clientDetail"
    >
      <div
        class="clientName"
      >
        bob
      </div>
      <div
        class="businessName"
      >
        <h1
          aria-label="business-name"
        >
          bobs business
        </h1>
        <h3>
          Business Name
        </h3>
      </div>
      <div
        class="infoWrapper"
      >
        <h3>
          <div
            aria-label="phone"
            class="contactDetails"
          >
            <section
              class="contactItem"
            >
              <img
                alt="phone icon"
                src="phoneSmall.png"
              />
              123
            </section>
            <section
              class="contactItem"
            >
              <img
                alt="email"
                src="emailSmall.png"
              />
              bob@bob.com
            </section>
          </div>
        </h3>
        <section
          class="projectSection"
        >
          <h2>
            Open Projects
          </h2>
          <section
            class="projectList"
          >
            <ul>
              <li>
                <a
                  href="/projects/1"
                >
                  <img
                    alt="palette"
                    src="paletteSmall.png"
                  />
                   
                  project1
                </a>
                <section
                  class="projectDescription"
                />
              </li>
              <li>
                <a
                  href="/projects/2"
                >
                  <img
                    alt="palette"
                    src="paletteSmall.png"
                  />
                   
                  project2
                </a>
                <section
                  class="projectDescription"
                />
              </li>
            </ul>
          </section>
        </section>
      </div>
      <section
        class="notesSection"
      >
        <h2>
          Notes
        </h2>
        <p
          aria-label="notes"
          class="notes"
        >
          yo bobs pretty cool
        </p>
      </section>
    </div>
  </main>
</div>
`;
